### DISCUSSION TITLE
Documents or paper list on constraints

### DISCUSSION THREAD

**Discussion by chenzhike110** [InteractiveComputerGraphics/PositionBasedDynamics#170] [Q&A]
I'm sorry to bother you, but I am new to PBD and your open source is extremly excellent. I wonder if you can provide a total document or a paper list on your providing constraints, because although your code is very clean, it remains difficult for me to understand ...

**Comment by janbender** [comment]
Unfortunately, not all constraints are documented in detail. However, you can find a complete list in the API documentation:

https://positionbaseddynamics.readthedocs.io/en/latest/api/file_Simulation_Constraints.h.html#classes
